backend/cms/models/page.py

```diff
@@ -36,24 +40,25 @@ def get_tree_view(cls):
     def depth(self):
         return len(self.get_ancestors())
 
+
     class MPTTMeta:
         order_insertion_by = ['order']
 
 
 class PageTranslation(models.Model):
+    page = models.ForeignKey(Page, related_name='page_translations')
+    permalink = models.CharField(max_length=60)
     STATUS = (
         ('draft', 'Entwurf'),
         ('review', 'Ausstehender Review'),
         ('public', 'Veröffentlicht'),
     )
-    status = models.CharField(max_length=10, choices=STATUS, default='draft')
     title = models.CharField(max_length=250)
+    status = models.CharField(max_length=10, choices=STATUS, default='draft')
     text = models.TextField()
-    permalink = models.CharField(max_length=60)
-    language = models.CharField(max_length=2)
+    language = models.ForeignKey(Language)
     version = models.PositiveIntegerField(default=0)
     active_version = models.BooleanField(default=False)
```

Comment:
what exactly does it mean for a version of a translation to be active? That it's the latest version? That it's the version that is going to be delivered via API? How does the status of a translation affect this?

Reply:
This isn't related to the translation status, but for the revision process. So we will have the same page with different revision states. But this field should only be true for one version, that is delivered via the API.

Reply:
Hmm... Not sure if I completely understood it: One page has several revisions and one revision (aka version) has several actual translations. Here, we have translations and revision packed into one model PageTranslation.
What happens if the active version has missing translations? Do we fallback to another version? If so, which one?

Reply:
> One page has several revisions

A page translation has several revision

> What happens if the active version has missing translations? Do we fallback to another version for the missing languages? If so, which one?

We should ensure that the active version is always a valid translation (if this is not the case, i.e. if there is a change to another language), then we inform the user about the need to update in the backend.

Reply:
Hmm not sure what you mean by invalid translation... is it an invalid state, in which there's a language, that the Site supports, but doesn't have any active PageTranslation revision for that page?

Also: How can we track changes in the PageTranslations of the source language, so that we can inform the user that the other PageTranslations are outdated? Are the version numbers of the source-language PageTranslation transferred to the destination-languages PageTranslations once translated? And does a version-number increment on any change? Is there a possibility for the authors to change the text without changing the version? Do we always create a new PageTranslation object for every change?

Reply:
> Hmm not sure what you mean by invalid translation...

Sorry, my choice of words was quite confusing. Forget about the "invalid".

> Also: How can we track changes in the PageTranslations of the source language, so that we can inform the user that the other PageTranslations are outdated?

Good point. I think synchronizing the version numbers could become very complicated and error-prone. As in the current system we should mark one language as main language and save for every translation the version of the main language that the translation was derived from.

> And does a version-number increment on any change?

Yes, in my opinion.

> Is there a possibility for the authors to change the text without changing the version?

I think the version is just a internal state. But there should be the possibility for the user to change the page translation in the main language without demanding the other translations to adjust. This functionality is also available in the current CMS.
So we may need a field to mark the version (page translation) in the main language as a minor change that has no effect on the other translations.

> Do we always create a new PageTranslation object for every change?

I think so. Maybe we can implement the minor changes without the previously mentioned flag/field and change the current version directly. This would prevent the creation of a new PageTranslation object.

Reply:
IMHO the model looks good so far. To clarify: "active" means that the text has been reviewed (whatever that means is up to the Verwalter / users). The latest "active" version is the page shown on the API / to the public. Correct?

Reply:
Ah and one other thing: do we need a "minor_edit" flag?

In Wordpress the minor edit flag for the main language implies that the translations are not outdated. This is used, for example, if a typo was fixed in the main language. Fixing typos does not (usually) invalidate translations.

Reply:
> Ah and one other thing: do we need a "minor_edit" flag?

Either we add a `minor_edit` flag to the PageTranslation or we just change the current PageTranslation for minor edits (losing traceability and a proper history of the translation). I vote for the flag.

> To clarify: "active" means that the text has been reviewed (whatever that means is up to the Verwalter / users). The latest "active" version is the page shown on the API / to the public. Correct?

No. The `active` flag is true, if (and only if) the PageTranslation should be delivered via API. Thus I consider renaming it to a `public` flag.
The `active` (`public`) flag can only be true for one PageTranslation per page & language.
(In formula):
![equation](https://latex.codecogs.com/svg.latex?%5Cbegin%7Barray%7D%7Bc%7D%20%5Cforall%20t_1%2Ct_2%5Cin%5Ctext%7BPageTranslation%7D%3A%20%5C%5C%20t_1.%5Ctext%7Bpage%7D%20%3D%20t_2.%5Ctext%7Bpage%7D%5C%20%5Cwedge%5C%20t_1.%5Ctext%7Blanguage%7D%3Dt_2.%5Ctext%7Blanguage%7D%5C%20%5Cwedge%5C%20t_1.%5Ctext%7Bactive%7D%20%3D%201%20%3D%20t_2.%5Ctext%7Bactive%7D%20%5CRightarrow%20t_1%20%3D%20t_2%20%5Cend%7Barray%7D)

The status of a PageTranslation on the other hand is only there for the revision process. Therefore I consider renaming the status choices from (draft, review, public) to (draft, in-review, reviewed).